src/yup.ts

```diff
@@ -1,44 +1,46 @@
 import { Resolver, transformToNestObject } from 'react-hook-form';
-import { FieldError, FieldErrors } from 'react-hook-form/dist/types/form';
-import { DeepMap } from 'react-hook-form/dist/types/utils';
 import Yup from 'yup';
 
-const parseErrorSchema = <TFieldValues extends Record<string, any>>(
+const parseErrorSchema = (
   error: Yup.ValidationError,
   validateAllFieldCriteria: boolean,
-): DeepMap<TFieldValues, FieldError> =>
+) =>
   Array.isArray(error.inner) && error.inner.length
-    ? error.inner.reduce<FieldErrors<TFieldValues>>(
-        (previous, { path, message, type }) => {
-          const previousPath = previous[path] as FieldErrors | undefined;
-          const previousTypes = validateAllFieldCriteria
-            ? previousPath?.types || []
-            : {};
-          return {
-            ...previous,
-            [path]: {
-              ...(previous[path] || {
-                message,
-                type,
-                types: previousTypes,
-              }),
-              types: {
-                ...previousTypes,
-                [type]: validateAllFieldCriteria
-                  ? [message, ...(previousTypes[type] || [])]
-                  : message,
-              },
-            },
-          };
-        },
+    ? error.inner.reduce(
+        (previous: Record<string, any>, { path, message, type }) => ({
+          ...previous,
+          ...(path
+            ? {
+                [path]: {
+                  ...(previous[path] || {
+                    message,
+                    type,
+                  }),
+                  ...(validateAllFieldCriteria
+                    ? {
+                        types: {
+                          ...((previous[path] && previous[path].types) || {}),
```

Comment:
store in variable to avoid redundancy maybe with line 22 and 24. Not mandatory.

Reply:
Not mandatory too to save few bytes as @bluebill tell me before

Reply:
Thanks. I will fix 👍

Reply:
done.